jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/nodejs/NodeJSRuntimeGroup.kt

```diff
@@ -0,0 +1,27 @@
+package software.aws.toolkits.jetbrains.services.lambda.nodejs
+
+import com.intellij.lang.javascript.JavaScriptSupportLoader
+import com.intellij.lang.javascript.JavascriptLanguage
+import com.intellij.openapi.projectRoots.Sdk
+import software.amazon.awssdk.services.lambda.model.Runtime
+import software.aws.toolkits.jetbrains.services.lambda.RuntimeGroupInformation
+
+class NodeJSRuntimeGroup : RuntimeGroupInformation {
+    override val runtimes: Set<Runtime> = setOf(
+            Runtime.NODEJS,
+            Runtime.NODEJS8_10,
+            Runtime.NODEJS6_10,
+            Runtime.NODEJS4_3,
+            Runtime.NODEJS4_3_EDGE
+    )
+
+    override val languageIds: Set<String> = setOf(
+            JavascriptLanguage.INSTANCE.id,
+            JavaScriptSupportLoader.TYPESCRIPT.id,
+            JavaScriptSupportLoader.FLOW_JS.id,
+            JavaScriptSupportLoader.ECMA_SCRIPT_6.id
+    )
+
+    // TODO we need module here, NodeJS doesn't use Sdk concept
+    override fun runtimeForSdk(sdk: Sdk): Runtime? = Runtime.NODEJS
```

Comment:
Should be safe to change to `runtimeForSdk(module: Module, sdk: Sdk)`, or is the SDK null completely in WebStorm?

Reply:
SDK is null in WebStorm. In other IDEs SDK may be not null, but it's not required for NodeJS plugin to function.

Reply:
Why don't we remove `runtimeForSdk` and replace it with `runtimeForModule(Module)` - those languages that do have a module and want the SDK to drive that can just use the `ModuleRootManager`. We may also need to introduce a `runtimeForProject(Project)` method.

There's two usages of the current `runtimeForSdk`:
* [`SamRunSettingsEditor`](https://github.com/aws/aws-toolkit-jetbrains/blob/9d98f62b9cf655b0fdd417666454d8eebe9911b9/jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunConfiguration.kt#L234) this only has `project` in scope, the `RunConfiguration` is not (currently) tied to a module, but we can probably infer it from the handler element.
* [`LambdaSamRunConfigurationProducer`](https://github.com/aws/aws-toolkit-jetbrains/blob/9d98f62b9cf655b0fdd417666454d8eebe9911b9/jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/LambdaSamRunConfigurationProducer.kt#L60) this *does* have module in scope, so we can replace its usage.

Reply:
Yep, that would work